File: llm_generated_code/Gemini/Python/Level1/Q17.py
def max_subarray_sum(arr):
    max_so_far = arr[0]
    max_ending_here = arr[0]
    start = 0
    end = 0
    s = 0

    for i in range(1, len(arr)):
        max_ending_here = max(arr[i], max_ending_here + arr[i])
        max_so_far = max(max_so_far, max_ending_here)
        if max_ending_here == arr[i]:
            s = i
        end = i
    print("Subarray:", arr[s:end+1])
    return max_so_far

File: llm_generated_code/Gemini/CPP/Level1/Q17.cpp
#include <iostream>
#include <vector>

using namespace std;

void findMaxSubarray(const vector<int>& arr) {
    int n = arr.size();
    int maxSum = arr[0], currentSum = arr[0];
    int start = 0, end = 0, s = 0;

    for (int i = 1; i < n; i++) {
        currentSum = max(arr[i], currentSum + arr[i]);
        if (currentSum > maxSum) {
            maxSum = currentSum;
            start = s;
            end = i;
        }
        if (currentSum < 0) {
            currentSum = 0;
            s = i + 1;
        }
    }

    cout << "Maximum sum: " << maxSum << endl;
    cout << "Subarray: ";
    for (int i = start; i <= end; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    vector<int> arr = {-2, -3, 4, -1, -2, 1, 5, -3};
    findMaxSubarray(arr);
    return 0;
}